pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -592,19 +592,23 @@ func (r *ClickHouseReader) GetRulesFromDB() (*[]model.RuleResponseItem, *model.A
 
 func (r *ClickHouseReader) GetRule(id string) (*model.RuleResponseItem, *model.ApiError) {
 
-	idInt, _ := strconv.Atoi(id)
+	idInt, err := strconv.Atoi(id)
+	if err != nil {
+		zap.S().Debug("Error in parsing param: ", err)
+		return nil, &model.ApiError{Typ: model.ErrorBadData, Err: err}
+	}
 
 	rule := &model.RuleResponseItem{}
 
 	query := fmt.Sprintf("SELECT id, updated_at, data FROM rules WHERE id=%d", idInt)
 
-	err := r.localDB.Get(rule, query)
+	err = r.localDB.Get(rule, query)
 
 	zap.S().Info(query)
 
 	if err != nil {
 		zap.S().Debug("Error in processing sql query: ", err)
-		return nil, &model.ApiError{Typ: model.ErrorInternal, Err: err}
+		return nil, &model.ApiError{Typ: model.ErrorNotFound, Err: err}
```

Comment:
Does non-nil err always mean no entry for rule?

Reply:
The lib returns an `error` object which needs to be parsed for text to generate different error codes. But then I looked into the API, it does not have much scope for error except for other internal errors. We could return `404` by matching the exact string and `500` for the rest of the errors.

Reply:
Should I update for the above?

Reply:
I think it's just fine to return 404. I just wanted to know if there are any other common errors. 

> it does not have much scope for error except for other internal errors

You are right. I just checked this https://pkg.go.dev/github.com/jmoiron/sqlx@v1.3.4#DB.Get

Reply:
@srikanthccv I have changed the function used to get data and added multiple checks for different error types.